plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -1115,6 +1115,7 @@ public boolean downloadToFile(String urlString, Plugin plugin, @CheckForNull Fil
                     List<URI> redirectLocations = context.getRedirectLocations();
                     // Expected to be an absolute URI
                     URI location = URIUtils.resolve(httphead.getURI(), target, redirectLocations);
+                    logVerbose(String.format("downloading %s from %s", plugin.getName(), location));
```

Comment:
there's probably a case for just moving this into the catch block and logging it all the time? not sure why it's verbose only for a failure case

Reply:
I like the idea of logging it on every failure so that users are more likely to report a mirror that is misbehaving.

Reply:
+1